src/Analyser/TypeSpecifier.php

```diff
@@ -681,14 +679,34 @@ public function specifyTypesInCondition(
 					throw new ShouldNotHappenException();
 				}
 
-				return array_reduce(
-					array_filter(
-						$expr->vars,
-						static fn (Expr $var) => $scope->issetCheck($var, static fn () => true),
-					),
-					fn (SpecifiedTypes $types, Expr $var) => $types->unionWith($this->specifyTypesInCondition($scope, $var, $context, $rootExpr)),
-					new SpecifiedTypes(),
-				);
+				$specifiedTypes = new SpecifiedTypes();
+				foreach ($expr->vars as $var) {
+					$isset = $scope->issetCheck($var, static fn () => true);
+
+					if ($isset !== true) {
+						continue;
+					}
+
+					$type = $scope->getType($var);
+					if (
+						!$type instanceof MixedType && !TypeCombinator::containsNull($type)
```

Comment:
`!$type instanceof MixedType && ` doesn't need to be there as `TypeCombinator::containsNull` will never return `true` for `MixedType`

Reply:
without this explicit `MixedType` check the 

```
function mixedIsset(mixed $m): void
{
	if (isset($m)) {
		assertType("mixed~null", $m);
	} else {
		assertType("null", $m);
	}
}
```

case will not succeed. I turned the conditions arround, maybe its more readable now.